src/sys/inotify.rs

```diff
@@ -213,33 +218,36 @@ impl InotifyInstance {
             )
         };
 
-        let bytes_left_in_buffer = buffer.len() - header_size;
-        assert!(bytes_left_in_buffer >= event.len as usize);
-
-        let bytes_consumed = header_size + event.len as usize;
-        let name = &buffer[header_size..bytes_consumed];
+        let name = match event.len {
+            0 => None,
+            _ => {
+                let ptr = unsafe { buffer.as_ptr().offset(header_size as isize) as *const c_char };
```

Comment:
According to the man page, `read` can return more than one event at a time.  If that happens, your code is going to ignore events past the first.